test.sh

```diff
@@ -146,7 +146,18 @@ GOBIN=${GOBIN:-$HOME/gopath/bin}
 #
 if [[ "$RUN" =~ "vet" ]] ; then
   start_context "test/vet"
-  run_and_comment go vet ./...
+  check_go_vet() {
```

Comment:
I think you've got it backwards.  It's `run` that calls `update_status`, which is what I'm after here -- causing failure when there's output, not just commenting.  For that to happen the argument to `run` needs to return non-zero.